qcodes/instrument/group_parameter.py

```diff
@@ -160,19 +163,19 @@ def __init__(self,
                                             Dict[str, Any]], None] = None,
                  separator: str = ','
                  ) -> None:
-        self.parameters = OrderedDict((p.name, p) for p in parameters)
+        self._parameters = OrderedDict((p.name, p) for p in parameters)
 
         for p in parameters:
-            p.group = self
+            p._group = self
 
         if len(set([p.root_instrument for p in parameters])) > 1:
             raise ValueError(
                 "All parameters should belong to the same instrument")
 
-        self.instrument = parameters[0].root_instrument
+        self._instrument = parameters[0].root_instrument
 
-        self.set_cmd = set_cmd
-        self.get_cmd = get_cmd
+        self._set_cmd = set_cmd
+        self._get_cmd = get_cmd
 
         if get_parser:
             self.get_parser = get_parser
```

Comment:
Hmn `get_parser` and `set_parser` should mayby also be private perhaps with a read only property. However they are exposed as attributes on `Parameter` so I have left them the same here

Reply:
> I have left them the same here

good